CAR-17: Add support for asserting JSONs equality in tests

majagabajaga opened the pull request:
- Added support for loading JSONs as JsonElement from resources
- Added Asserting of equality of JSONs
- Added UTF-8 as default encoding
- Added commons-io as dependencies

kagbepreved committed:
CAR-17: Add support for asserting JSONs equality in tests

- Added support for loading JSONs as JsonElement from resources
- Added Asserting of equality of JSONs
- Added UTF-8 as default encoding
- Added commons-io as dependencies

kagbepreved committed:
CAR-17: Add support for asserting JSONs equality in tests

Added new lines at the end of JSON files.

kagbepreved committed:
CAR-17: Add support for asserting JSONs equality in tests

JsonAssert:
- simplify parsing of JSON files
- avoid code duplication on assertions
- enhance assertion to handle different order of properties in json
files
- add more properties to the files being compared
- add private constructor to Constants

kagbepreved committed:
CAR-17: Add support for asserting JSONs equality in tests

Added the .gitattributes file.
Force all *.txt files to have unix-style eol.

mironuk commented on .gitattributes:
```diff
@@ -0,0 +1 @@
+*.txt	eof=lf
```
There shoud be new line after last line.

majagabajaga replied:
Added the new line character.

mironuk commented on src/test/java/carservice/test/AssertExt.java:
```diff
@@ -13,37 +13,36 @@
 
-        JsonReader jsonReader1 = new JsonReader(r1);
-        JsonReader jsonReader2 = new JsonReader(r2);
+        JsonReader expectedJsonReader = new JsonReader(r1);
```
Seems like these two lines can be moved to common method also.

majagabajaga replied:
Done.

mironuk commented on src/test/java/carservice/test/testdata/f1.json:
```diff
@@ -1,3 +1,8 @@
+        "color": "black",
+        "meowType": "annoying"
+    },
+    "cat": "Barseeq"
```
Show stopper error: "Barseeq" should be "Barseek". Just kidding. :)

majagabajaga replied:
Donen't.

mironuk commented on src/test/java/carservice/test/AssertExtTest.java:
```diff
@@ -0,0 +1,39 @@
+    @Test
+    public void givenEqualJsons_whenAssertJsonsEqual_thenSuccess() {
+        // Arrange
+        String json1 = "{ \"meow\": \"meow\" }";
```
I think there should be string examples from f1.json and f2.json (full json with shuffled fields)

majagabajaga replied:
Done.

kagbepreved committed:
CAR-17: Add support for asserting JSONs equality in tests

- AssertExt: Improved code reuse
- AssertExtTest: Use more complicated JSON files to test the
assertJsonsEqual() method.
-.gitattributes: added EOL
